### ISSUE TITLE
Release 2.3 BBB BTRFS doesn't boot on PocketBeagle

### ISSUE THREAD

**Issue by jadonk** [FalconChristmas/fpp#451]
```
U-Boot SPL 2018.09-00002-gd5b4c4b656 (Oct 19 2018 - 14:16:02 -0500)
Trying to boot from MMC1
Loading Environment from EXT4... ** File not found /boot/uboot.env **

** Unable to read "/boot/uboot.env" from mmc0:1 **


U-Boot 2018.09-00002-gd5b4c4b656 (Oct 19 2018 - 14:16:02 -0500), Build: jenkins-github_Bootloader-Builder-84

CPU  : AM335X-GP rev 2.1
I2C:   ready
DRAM:  512 MiB
No match for driver 'omap_hsmmc'
No match for driver 'omap_hsmmc'
Some drivers were not found
Reset Source: Power-on reset has occurred.
RTC 32KCLK Source: Internal.
MMC:   OMAP SD/MMC: 0, OMAP SD/MMC: 1
Loading Environment from EXT4... ** File not found /boot/uboot.env **

** Unable to read "/boot/uboot.env" from mmc0:1 **
Model: BeagleBoard.org PocketBeagle
<ethaddr> not set. Validating first E-fuse MAC
Net:   usb_ether
Press SPACE to abort autoboot in 2 seconds
board_name=[A335PBGL] ...
switch to partitions #0, OK
mmc0 is current device
SD/MMC found on device 0
switch to partitions #0, OK
mmc0 is current device
Scanning mmc 0:1...
gpio: pin 56 (gpio 56) value is 0
gpio: pin 55 (gpio 55) value is 0
gpio: pin 54 (gpio 54) value is 0
gpio: pin 53 (gpio 53) value is 1
switch to partitions #0, OK
mmc0 is current device
gpio: pin 54 (gpio 54) value is 1
Checking for: /uEnv.txt ...
2145 bytes read in 7 ms (298.8 KiB/s)
gpio: pin 55 (gpio 55) value is 1
Loaded environment from /uEnv.txt
Importing environment from mmc ...
Checking if uenvcmd is set ...
Checking if client_ip is set ...
Checking for: /boot.scr ...
Checking for: /boot/boot.scr ...
Checking for: /boot/uEnv.txt ...
gpio: pin 55 (gpio 55) value is 1
2145 bytes read in 11 ms (190.4 KiB/s)
Loaded environment from /boot/uEnv.txt
Checking if uname_r is set in /boot/uEnv.txt...
gpio: pin 56 (gpio 56) value is 1
Running uname_boot ...
loading /boot/vmlinuz-4.14.71-bone17 ...
10285568 bytes read in 663 ms (14.8 MiB/s)
debug: [enable_uboot_overlays=1] ...
debug: [enable_uboot_cape_universal=1] ...
debug: [uboot_base_dtb_univ=] ...
loading /boot/dtbs/4.14.71-bone17/am335x-pocketbeagle.dtb ...
129638 bytes read in 32 ms (3.9 MiB/s)
uboot_overlays: [fdt_buffer=0x60000] ...
uboot_overlays: loading /lib/firmware/AM335X-PRU-UIO-00A0.dtbo ...
883 bytes read in 103 ms (7.8 KiB/s)
debug: [console=ttyO0,115200n8 root=/dev/mmcblk0p2 ro rootfstype=btrfs coherent_pool=1M net.ifnames=0 quiet] ...
debug: [bootz 0x82000000 - 88000000] ...
## Flattened Device Tree blob at 88000000
   Booting using the fdt blob at 0x88000000
   reserving fdt memory region: addr=88000000 size=80000
   Loading Device Tree to 8ff7d000, end 8fffffff ... OK

Starting kernel ...

[    0.001094] timer_probe: no matching timers found
[    0.515467] dmi: Firmware registration failed.
[    0.704124] wkup_m3_ipc 44e11324.wkup_m3_ipc: could not get rproc handle
[    0.870010] omap_voltage_late_init: Voltage driver support not added
[    0.924591] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    0.932910] CPU: 0 PID: 1 Comm: swapper Not tainted 4.14.71-bone17 #1
[    0.939374] Hardware name: Generic AM33XX (Flattened Device Tree)
[    0.945534] [<c0111360>] (unwind_backtrace) from [<c010d910>] (show_stack+0x20/0x24)
[    0.953322] [<c010d910>] (show_stack) from [<c0dbca2c>] (dump_stack+0x24/0x28)
[    0.960593] [<c0dbca2c>] (dump_stack) from [<c013a048>] (panic+0xec/0x26c)
[    0.967508] [<c013a048>] (panic) from [<c1401550>] (mount_block_root+0x1b8/0x29c)
[    0.975027] [<c1401550>] (mount_block_root) from [<c1401774>] (mount_root+0x140/0x15c)
[    0.982980] [<c1401774>] (mount_root) from [<c14018f0>] (prepare_namespace+0x160/0x1a8)
[    0.991022] [<c14018f0>] (prepare_namespace) from [<c14010b8>] (kernel_init_freeable+0x1d0/0x1e4)
[    0.999944] [<c14010b8>] (kernel_init_freeable) from [<c0dd2a18>] (kernel_init+0x18/0x128)
[    1.008255] [<c0dd2a18>] (kernel_init) from [<c0108e48>] (ret_from_fork+0x14/0x2c)
[    1.015876] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.025077] musb-hdrc musb-hdrc.1: VBUS_ERROR in a_wait_vrise (81, <SessEnd), retry #1, port1 00000104
[    1.034959] remoteproc remoteproc0: remote processor wkup_m3 is now up
[    1.034976] wkup_m3_ipc 44e11324.wkup_m3_ipc: CM3 Firmware Version = 0x193
[    1.059491] mmc0: host does not support reading read-only switch, assuming write-enable
[    1.081261] mmc0: new high speed SDHC card at address aaaa
[    1.094868] mmcblk0: mmc0:aaaa SS08G 7.40 GiB
[    1.108380]  mmcblk0: p1 p2
[    1.177524] musb-hdrc musb-hdrc.1: VBUS_ERROR in a_wait_vrise (81, <SessEnd), retry #2, port1 00000104
[    1.329930] musb-hdrc musb-hdrc.1: VBUS_ERROR in a_wait_vrise (81, <SessEnd), retry #3, port1 00000104
[    1.482356] musb-hdrc musb-hdrc.1: VBUS_ERROR in a_wait_vrise (80, <SessEnd), retry #3, port1 0008010c
[   52.846802] random: crng init done
```

And it just sits there.

**Comment by JonB256** [comment]
v2.3 is actually v2.2 with a few updates and released as a new image. Not your imagination.

**Comment by dkulp** [comment]
As an FYI, I cannot reproduce this why my PocketBeagles using the btrfs image.  However, the rootwait wouldn't harm anything so it does make sense to add.  I'll add that for the 2.4 images

**Comment by dkulp** [comment]
Will be fixed for 2.4
